### PR TITLE
fix: delete dimensions when deleting table

### LINKED ISSUE #257: Deleting the observation table should delete related dimensions

### PR DESCRIPTION
**Pull request by lucafurrer**
Deletes related dimensions when a table is deleted

fix #257

### CODE REVIEW

File: apis/core/test/domain/table/delete.test.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import { TestResourceStore } from '../../support/TestResourceStore'
+import { NamedNode } from 'rdf-js'
+import type * as DimensionMetadataQueries from '../../../lib/domain/queries/dimension-metadata'
+import type * as TableQueries from '../../../lib/domain/queries/table'
+import type * as ColumnMappingQueries from '../../../lib/domain/queries/column-mapping'
+import '../../../lib/domain'
+import { deleteTable } from '../../../lib/domain/table/delete'
+import { ColumnMapping, Table } from '@cube-creator/model'
+
+describe('domain/table/delete', () => {
```
Comment (tpluscode) [inline apis/core/test/domain/table/delete.test.ts:18]: Is this test suite missing the cases when deleted table removes (or not) the dimensions?

File: apis/core/lib/domain/dimension/DimensionMetadataCollection.ts [hunk 2]
```diff
@@ -57,8 +57,10 @@ export default function Mixin<Base extends Constructor<Omit<DimensionMetadataCol
     addDimensionMetadata(params: CreateColumnMetadata): DimensionMetadata.DimensionMetadata {
       const identifier = params.csvMapping.createIdentifier(params.columnMapping.targetProperty)
 
-      if (this.hasPart.some((dimMeta) => dimMeta.about === identifier)) {
-        throw new Error(`Dimension Metadata with identifier ${identifier.value} already exists`)
+      const existingDim = this.hasPart.find(dimMeta => dimMeta.about.equals(identifier))
+
+      if (existingDim) {
+        return existingDim
```
Comment (tpluscode) [inline apis/core/lib/domain/dimension/DimensionMetadataCollection.ts:63]: codecov indicates that these lines are untested

File: apis/core/test/domain/table/delete.test.ts (hunk 1 above)
Comment (lucafurrer) [reply apis/core/test/domain/table/delete.test.ts:18]: You have the check that the dimension is deleted at line 143, I add the reverse check

File: apis/core/lib/domain/dimension/DimensionMetadataCollection.ts (hunk 2 above)
Comment (lucafurrer) [reply apis/core/lib/domain/dimension/DimensionMetadataCollection.ts:63]: added a test

**Review by tpluscode — APPROVED** [review]

**Pull request merged**